lib/PHPExif/Reader/Reader.php

```diff
@@ -100,7 +108,10 @@ public static function factory($type)
                 );
                 break;
         }
-        return new $classname($adapter);
+        $result = new $classname();
```

Comment:
Keeping previous comment in mind, this should be reverted to ``return new $classname($adapter);``